### PR TITLE
Fix: Fix a bug when the hostname label of a node does not match the node name, pods bound to a PV with nodeAffinity using the hostname may be scheduled to the wrong node or experience scheduling failures.

### PR DESCRIPTION
**Pull request by dongjiang1989**
cherry pick https://github.com/kubernetes/kubernetes/pull/127584 to volcano
ref: https://github.com/kubernetes/kubernetes/pull/127584

### CODE REVIEW

File: pkg/scheduler/capabilities/volumebinding/volume_binding.go [hunk 1]
```diff
@@ -343,13 +343,6 @@ func (pl *VolumeBinding) PreFilter(ctx context.Context, state *framework.CycleSt
 		status.AppendReason("pod has unbound immediate PersistentVolumeClaims")
 		return nil, status
 	}
-	// Attempt to reduce down the number of nodes to consider in subsequent scheduling stages if pod has bound claims.
```
Comment (Monokaix) [inline pkg/scheduler/capabilities/volumebinding/volume_binding.go:346]: Why is this removed and is it synced from k8s？

Comment (dongjiang1989) [reply pkg/scheduler/capabilities/volumebinding/volume_binding.go:346]: Yes. ref：

File: pkg/scheduler/capabilities/volumebinding/volume_binding_test.go [hunk 2]
```diff
@@ -148,9 +147,7 @@ func TestVolumeBinding(t *testing.T) {
 					v1.LabelHostname: {"node-a"},
 				}).PersistentVolume,
 			},
-			wantPreFilterResult: &framework.PreFilterResult{
```
Comment (Monokaix) [inline pkg/scheduler/capabilities/volumebinding/volume_binding_test.go:151]: Please sync all changed files: )

**Comment by dongjiang1989** [comment]
Thanks @Monokaix 
Please re-check it.

**Comment by Monokaix** [comment]
This pr https://github.com/volcano-sh/apis/pull/146 has been merged, and please update the go mod: )

**Comment by dongjiang1989** [comment]
> This pr [volcano-sh/apis#146](https://github.com/volcano-sh/apis/pull/146) has been merged, and please update the go mod: )

Maybe use apis new tag? @Monokaix

**Comment by Monokaix** [comment]
I think it's better to split it into two commits, one is for auo gerneate yaml and another one is fix volume binding err: )
